### ISSUE TITLE
Gradle's JVM probing logic fails when using JDK with wrong page size on ARM

### ISSUE THREAD

**Issue by breskeby** [gradle/gradle#16318]
Due to this open [JDK bug](https://bugs.openjdk.java.net/browse/JDK-8236847)  Gradle JVM probing fails as there's unexpected output with this additional line:

```
[0.015s][error][cds] Unable to map CDS archive -- os::vm_allocation_granularity() expected: 4096 actual: 65536 
``` 

The probing logic in Gradle should be more resilient since this "error" isn't actually fatal.

### Expected Behavior

Gradle JVM probing does not fail in the described scenario

### Current Behavior

Gradle JVM probing fails with the following output:

```

Unable to determine version for JDK located at /var/lib/jenkins/.gradle/caches/transforms-3/741c1b0a94f5a1be265ba151b1841ee6/transformed/linux-15.0.1-aarch64.tar.gz. Reason: Unexpected command output: 
[0.003s][error][cds] Unable to map CDS archive -- os::vm_allocation_granularity() expected: 4096 actual: 65536
/var/lib/jenkins/.gradle/caches/transforms-3/741c1b0a94f5a1be265ba151b1841ee6/transformed/linux-15.0.1-aarch64.tar.gz
15.0.1
AdoptOpenJDK
aarch64
OpenJDK 64-Bit Server VM
15.0.1+9
OpenJDK Runtime Environment


at org.gradle.internal.jvm.inspection.DefaultJvmVersionDetector.validate(DefaultJvmVersionDetector.java:59)  
at org.gradle.internal.jvm.inspection.DefaultJvmVersionDetector.getVersionFromJavaHome(DefaultJvmVersionDetector.java:52)
at org.gradle.internal.jvm.inspection.DefaultJvmVersionDetector.getJavaVersion(DefaultJvmVersionDetector.java:48) |  
at org.gradle.api.tasks.testing.Test.getJavaVersion(Test.java:273)

``` 

### Context

We run into this when running our builds on `aarch64` systems. 
 

### Steps to Reproduce 

- Setup linux aarch64 System
- Install AdoptJDK 15.0.1+9 AARCH64
- `EXPORT RUNTIME_JAVA_HOME=<PATH TO ADOPTJDK 15>
- clone [elasticsearch](https://github.com/elastic/elasticsearch/)
- checkout master
- run `./gradlew  :x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest `

### Environment

```
Operating system | Linux 4.18.0-240.1.1.el8_3.aarch64 |  
-- | -- | --
CPU cores | 32 cores |  
Max Gradle workers | 32 workers |  
Java runtime | AdoptOpenJDK OpenJDK Runtime Environment 15.0.2+7 |  
Java VM | AdoptOpenJDK OpenJDK 64-Bit Server VM 15.0.2+7 (mixed mode) |  
Max JVM memory heap size | 29.9 GiB |  
Locale | English (United States) |  
Default charset | UTF-8 |  
Username | jenkins |  

```

**Comment by droberts195** [comment]
One extra piece of information if you try to reproduce this is that the error only gets printed if the linux-aarch64 machine you reproduce on has a different page size to the machine that your JDK was built on.

On linux-aarch64 RHEL/CentOS distributions are built with a page size of 64KB and Ubuntu/Debian/SLES distributions are built with a page size of 4KB.

We see the problem with AdoptJDK 15.0.1+9 AARCH64 on CentOS 8, which means that AdoptJDK 15.0.1+9 AARCH64 must have been built on a distribution that was built with a 4KB page size.

So if you try to reproduce with AdoptJDK 15.0.1+9 AARCH64 you will need to do this on RHEL or CentOS.  If you try to reproduce with AdoptJDK 15.0.1+9 AARCH64 on Ubuntu, Debian or SLES then you will not see the problem.

**Comment by bmuskalla** [comment]
Thanks for all the details on this, very appreciated!

**Comment by breskeby** [comment]
Are there plans to fix this for 7.0?
